File: factory/AmericanExpress.java
package factory;

public class AmericanExpress extends InternationalPaymentMethod {
    public AmericanExpress(String cardHolder, String cardNumber, String cvv, String expiryDate) {
        super(cardHolder, cardNumber, cvv, expiryDate);
    }

    @Override
    public void authorizePayment() {
        System.out.println("Authorizing american express card....");
    }

    @Override
    public void startMoneyTransfer() {
        System.out.println("Start money transferring for a american express card....");
    }

    @Override
    public void calculatePaymentFees(double amount) {
        System.out.println("Calculating american express card payment fees....");
    }
}

File: typescript/patterns/factory/american-express.ts
import { InternationalPaymentMethod } from "./international-payment-method";

export class AmericanExpress extends InternationalPaymentMethod {
    constructor(
      cardHolder: string,
      cardNumber: string,
      cvv: string,
      expiryDate: string
    ) {
      super(cardHolder, cardNumber, cvv, expiryDate);
    }
  
    authorizePayment(): void {
      console.log("Authorizing American Express card....");
    }
  
    startMoneyTransfer(): void {
      console.log("Start money transferring for an American Express card....");
    }
  
    calculatePaymentFees(amount: number): void {
      console.log("Calculating American Express card payment fees....");
    }
  }